secboot/secboot_sb_test.go

```diff
@@ -1757,18 +1702,14 @@ func (s *secbootSuite) TestUnlockVolumeUsingSealedKeyIfEncryptedFdeRevealKeyV2Mo
 	}
 
 	activated := 0
-	restore = secboot.MockSbActivateVolumeWithKeyData(func(volumeName, sourceDevicePath string, keyData *sb.KeyData, options *sb.ActivateVolumeOptions) (sb.SnapModelChecker, error) {
+	restore = secboot.MockSbActivateVolumeWithKeyData(func(volumeName, sourceDevicePath string, authRequestor sb.AuthRequestor, kdf sb.KDF, options *sb.ActivateVolumeOptions, keys ...*sb.KeyData) error {
+		c.Check(options.Model, Equals, fakeModel)
 		activated++
-		modChecker := &mockSnapModelChecker{mockIsAuthorized: false}
-		return modChecker, nil
-	})
-	defer restore()
-
-	deactivated := 0
-	restore = secboot.MockSbDeactivateVolume(func(volumeName string) error {
-		deactivated++
-		c.Check(volumeName, Equals, "device-name-random-uuid-for-test")
-		return nil
+		// XXX: remove entire test as it now only tests mocks? The
+		// new secboot code checks for the model internally in
+		// ActivateVolumeWIthKeyData (before it expecting a
+		// model-checker that we provided)
+		return fmt.Errorf("cannot unlock volume: model %s/%s not authorized", fakeModel.AuthorityID(), fakeModel.DisplayName())
```

Comment:
I also agree to remove this test completely. This is secboot behavior.

Reply:
I have merged the two tests into one since they were both testing error from activation.